tests/lib/state_test.py

```diff
@@ -138,3 +140,121 @@ def _extra_state(self):
                 {'name': 'eth1', 'state': 'up', 'type': 'unknown'}
             ]
         })
+
+
+class TestAssertRouteState(object):
+    def test_hash_unique(self):
+        routes = _get_mixed_test_routes()
+        assert (hash(state.RouteEntry(routes[0])) ==
+                hash(state.RouteEntry(routes[0])))
+
+    def test_obj_unique(self):
+        routes = _get_mixed_test_routes()
+        route = routes[0]
+        new_route = copy.deepcopy(routes[0])
+        assert state.RouteEntry(route) == state.RouteEntry(new_route)
+        assert state.RouteEntry(route) != state.RouteEntry(routes[1])
```

Comment:
I think this checks identity and not content.
I suspect this one will probably pass (when it should not):
`assert state.RouteEntry(route) != state.RouteEntry(new_route)`